projects/extension/sql/idempotent/901-semantic-catalog-set.sql

```diff
@@ -0,0 +1,137 @@
+--FEATURE-FLAG: text_to_sql
+
+-------------------------------------------------------------------------------
+-- add_sql_example
+create or replace function ai.add_sql_example
```

Comment:
I don't quite understand end-to-end how this is intended to be used. My intuition tells me that you would want to associate example SQL with some other objects.

Reply:
In the simplest case:

1. embed the user's question
2. do a similarity search on the example SQL descriptions
3. include the SQL + description as context in the prompt to generate the SQL

But yeah, that might not be good enough. I've thought about associating it with db objects that the example query references. To do that though, we either have to parse the query, generate a query plan with explain and parse that, or ask an LLM to identify the objects referenced. I'm not going to attempt that until it seems necessary

Reply:
unless, you require the user to supply a list of associated database objects when adding the example